### PR TITLE
Add Web Monetization

### PR DESCRIPTION
**Pull request by thomasbnt**
**[Awesome Web Monetization repository](https://github.com/thomasbnt/awesome-web-monetization#readme)**

**[Web Monetization](https://webmonetization.org) is becoming more and more used on the Internet. It would be interesting for people to turn to a new payment method for websites such as blogs, videos, newsletters or shared photos whether it is monetized according to the time spent reading/watching. And all this natively on your computer.**

___

Checked PR's:

#1800
#1811 
#1795 

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration.
- You have read and understood the [instructions for creating a list](create-list.md).
- This pull request has a title in the format `Add Name of List`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- Your entry should be added at the bottom of the appropriate category.
- The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it.
- Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- The default branch should be named [`main`, not `master`](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/).
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- Non-generated Markdown file in a GitHub repo.
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate. Please search for existing submissions.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
- Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=master&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name or text to the readme. GitHub already shows the license name at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- Has [contribution guidelines](awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Doesn't include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

### CODE REVIEW

**Comment by thomasbnt** [comment]
Oh and, unicorn ! :unicorn:

**Comment by thomasbnt** [comment]
Linked to this issue https://github.com/thomasbnt/awesome-web-monetization/issues/4

File: readme.md [hunk 1]
```diff
@@ -654,6 +654,7 @@
 - [Prometheus](https://github.com/roaldnefs/awesome-prometheus#readme) - Open-source monitoring system.
 - [Homematic](https://github.com/homematic-community/awesome-homematic#readme) - Smart home devices.
 - [Ledger](https://github.com/sfischer13/awesome-ledger#readme) - Double-entry accounting on the command-line.
+- [Web Monetization](https://github.com/thomasbnt/awesome-web-monetization) - Web service that allows you to send money directly in your browser.
```
Comment (sindresorhus) [inline readme.md:657]: This does not make it clear that it's an open standard. It makes it sound like it's a paid API.

**Comment by sindresorhus** [comment]
Thanks for making an Awesome list! 🙌

It looks like you didn't read the [guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) closely enough. I noticed multiple things that are not followed. Try going through the list point for point to ensure you follow it. I spent a lot of time creating the guidelines so I wouldn't have to comment on common mistakes, and rather spend my time improving Awesome.

**Comment by thomasbnt** [comment]
> [Requirements](https://github.com/sindresorhus/awesome/blob/main/pull_request_template.md#requirements-for-your-awesome-list) no. 4 indicates to rename/move the default branch from `master` to `main`.
> > - The default branch should be named main, not master.
> 
> You can follow the [steps here](https://www.hanselman.com/blog/EasilyRenameYourGitDefaultBranchFromMasterToMain.aspx) on how to rename this.

Ooops yeah I forgotten that on my repository

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/awesome/commit/7acb34c61dbac2cbc47d8438eb57aa29d79bc650

**Comment by Potherca** [comment]
@sindresorhus We ran the Awesome Linter, which passed... Is there anything that could be added there to make the issues we have here more clear? If yes (as I know you are very bussssssy), I don't mind doing the work to put 'm in...

**Comment by thomasbnt** [comment]
@mhxion Thanks for the post to edit that. The default branch is `main` now!

File: readme.md (hunk 1 above)
Comment (thomasbnt) [reply readme.md]: Edited with [`f078080` (#1812)](https://github.com/sindresorhus/awesome/pull/1812/commits/f078080f4659e4d95336f46ee65d1f4c263de2f5)

**Comment by thomasbnt** [comment]
> [7acb34c](https://github.com/sindresorhus/awesome/commit/7acb34c61dbac2cbc47d8438eb57aa29d79bc650)

Checked again the repositories, no errors for https://github.com/sindresorhus/awesome/pull/1800 and https://github.com/sindresorhus/awesome/pull/1795. 

Why do you not add a GitHub Action with awesome-lint and require to news Awesome repositories to use that before Pr's here ?

**Comment by thomasbnt** [comment]
> @thomasbnt From the requirements:
> 
> > Do not add the license name or text to the readme. GitHub already shows the license name at the top of the repo.
> 
> You need to remove the [license section](https://github.com/thomasbnt/awesome-web-monetization#license).

Cjecked

**Comment by thomasbnt** [comment]
Your [repository on Discord](https://github.com/mhxion/awesome-programming-discord) presentation is very cool ! Good job, and nice [with badges](https://github.com/mhxion/awesome-programming-discord/blob/main/badges.md#official-identification-badge) !

**Comment by thomasbnt** [comment]
@mhxion all is good ! Check with awesome lint for the repository. No more errors

**Comment by mhxion** [comment]
@thomasbnt I did run `awesome-lint`, no error as well. I don't know why `awesome-lint` didn't validate that, based on the requirement instead of
```md
- [svelte-monetization : A minimal and lightweight wrapper for the Web Monetization API](https://github.com/sorxrob/svelte-monetization) ![](assets/small_icons/svelte.png)
```
it needs to be
```md
- [svelte-monetization](https://github.com/sorxrob/svelte-monetization) : A minimal and lightweight wrapper for the Web Monetization API ![](assets/small_icons/svelte.png)
```
I guess @sindresorhus can verify this issue here if this isn't a strict requirement at all.

**Comment by sindresorhus** [comment]
The guidelines are the source of truth. The linter doesn't catch everything.

**Comment by thomasbnt** [comment]
> The guidelines are the source of truth. The linter doesn't catch everything.

Okay it's edited! 
:octopus:  https://github.com/thomasbnt/awesome-web-monetization/commit/9a5611a6c191d398a6b5e1f546372089096830ca

**Comment by sindresorhus** [comment]
There are still multiple things not follow from the guidelines. Please read through each item thoroughly.

**Comment by thomasbnt** [comment]
I've got [an error](https://github.com/thomasbnt/awesome-web-monetization/runs/835490035?check_suite_focus=true#step:5:12) but.. 

`readme.md:105:151
  ✖  105:151  List item description must end with proper punctuation  remark-lint:awesome-list-item`

there are that : 

```markdown
- [Monetization](https://github.com/KNawm/monetization) - A wrapper around the Web Monetization API to monetize apps. ![](assets/small_icons/dart.png) 
```

Why the asset could block and display this error knowing that for other resources no error is present?

File: readme.md [hunk 2]
```diff
@@ -654,6 +654,7 @@
 - [Prometheus](https://github.com/roaldnefs/awesome-prometheus#readme) - Open-source monitoring system.
 - [Homematic](https://github.com/homematic-community/awesome-homematic#readme) - Smart home devices.
 - [Ledger](https://github.com/sfischer13/awesome-ledger#readme) - Double-entry accounting on the command-line.
+- [Web Monetization](https://github.com/thomasbnt/awesome-web-monetization) - A free open web standard service that allows you to send money directly in your browser.
```
Comment (sindresorhus) [inline readme.md:657]: You have a trailing space and missing #readme

**Comment by sindresorhus** [comment]
> Includes a succinct description of the project/theme at the top of the readme. (Example)
✅ Mobile operating system for Apple phones and tablets.
✅ Prototyping interactive UI designs.
❌ Resources and tools for iOS development.
❌ Awesome Framer packages and tools.

Not done.

The following is not good enough:

> Web Monetization is a web service that allows you to send money directly in your browser. — If you want you know more, see :

And the description should be above the "Contents" section.

---

Everything under the `Resources` section, could be top-level sections.

---

Don't include empty sections.

---

There's very little content in the list.

**Comment by sindresorhus** [comment]
Bump

**Comment by thomasbnt** [comment]
Hello ! All is good, but you say there is little content at the moment. I prefer not to relaunch for the moment. a few changes have been made, contributors have played the game.💪🏼

**Comment by sindresorhus** [comment]
Tweet: https://twitter.com/awesome__re/status/1340349975943761920

**Pull request merged**